### DISCUSSION TITLE
Login/User Authentication

### DISCUSSION THREAD

**Discussion by akil-rails** [vuejs/vitepress#1293] [Q&A]
Some of my documentation needs to be accessible only on a need-to-know basis; and this requires some sort of authentication to be added to the site / certain section of the site. Any suggestions on how to do this?

AWS Amplify/Netlify allow for protecting the entire site with a shared user/password - that's not sufficient.

**Comment by brc-dd** [comment]
Its not built-in and likely won't be. You'll need some custom server running for this that can protect routes. Netlify has this built-in on paid plans: https://docs.netlify.com/visitor-access/password-protection/#custom-username-and-password-protection-through-basic-authentication

Also if you can deploy your site behind an apache/nginx server, you can configure basic auth there (https://developer.mozilla.org/en-US/docs/Web/HTTP/Authentication#restricting_access_with_nginx_and_basic_authentication).

**Comment by fiwalld** [comment]
Maybe you can try https://www.vpadmin.org
